avm/res/compute/gallery/image/main.bicep

```diff
@@ -256,3 +155,120 @@ type roleAssignmentType = {
+    architecture: 'x64' | 'Arm64'?
+
+    @sys.description('Optional. The description of this gallery image definition resource. This property is updatable.')
+    description: string?
+
+    @sys.description('Optional. Describes the disallowed disk types.')
+    disallowed: {
+      @sys.description('Required. A list of disk types.')
+      @metadata({ example: '''
+      [
+        'Standard_LRS'
+      ]''' })
+      diskTypes: string[]
+    }?
+
+    @sys.description('Optional. The end of life date of the gallery image definition. This property can be used for decommissioning purposes. This property is updatable.')
+    endOfLifeDate: string?
+
+    @sys.description('Optional. The Eula agreement for the gallery image definition.')
+    eula: string?
+
+    // features are skipped for now, as we prefer typed parameters
+
+    @sys.description('Optional. The hypervisor generation of the Virtual Machine. If this value is not specified, then it is determined by the securityType parameter. If the securityType parameter is specified, then the value of hyperVGeneration will be V2, else V1.')
+    hyperVGeneration: 'V1' | 'V2'?
+
+    @sys.description('Required. This is the gallery image definition identifier.')
+    identifier: {
+      @sys.description('Required. The name of the gallery image definition offer.')
+      publisher: string
+
+      @sys.description('Required. The name of the gallery image definition publisher.')
+      offer: string
+
+      @sys.description('Required. The name of the gallery image definition SKU.')
+      sku: string
+    }
+
+    @sys.description('Required. This property allows the user to specify whether the virtual machines created under this image are `Generalized` or `Specialized`.')
+    osState: 'Generalized' | 'Specialized'
+
+    @sys.description('Required. This property allows you to specify the type of the OS that is included in the disk when creating a VM from a managed image. Possible values are: `Windows`, `Linux`.')
+    osType: 'Linux' | 'Windows'
+
+    @sys.description('Optional. The privacy statement uri.')
+    privacyStatementUri: string?
+
+    @sys.description('Optional. Describes the gallery image definition purchase plan. This is used by marketplace images.')
+    purchasePlan: {
+      @sys.description('Required. The plan ID.')
+      name: string
+
+      @sys.description('Required. The product ID.')
+      product: string
+
+      @sys.description('Required. The publisher ID.')
+      publisher: string
+    }?
+
+    @sys.description('Optional. The properties describe the recommended machine configuration for this Image Definition. These properties are updatable.')
+    recommended: {
+      @sys.description('Optional. Describes the resource range (1-128 CPU cores). Defaults to min=1, max=4.')
+      vCPUs: resourceRangeType?
+
+      @sys.description('Optional. Describes the resource range (1-4000 GB RAM). Defaults to min=4, max=16.')
```

Comment:
Where are these defaults coming from?

Reply:
They are assigned when calling the image template from the main loop.

Reply:
That's for a child-module sadly not a good option as you must imagine that this module may at some point be deployed on it's own where the defaults won't apply. But - now that you split the UDT I can add some alternatives :)